src/python/pants/java/nailgun_protocol.py

```diff
@@ -237,10 +239,13 @@ def isatty_to_env(cls, stdin, stdout, stderr):
     :param file stderr: The stream to check for stderr tty capabilities.
     :returns: A dict containing the tty capability environment variables.
     """
-    fds = (stdin, stdout, stderr)
-    return {
-      cls.TTY_ENV_TMPL.format(fd_id): bytes(int(fd.isatty())) for fd_id, fd in enumerate(fds)
-    }
+    def gen_env_vars():
+      for fd_id, fd in enumerate((stdin, stdout, stderr)):
+        is_atty = fd.isatty()
+        yield (cls.TTY_ENV_TMPL.format(fd_id), bytes(int(is_atty)))
+        if is_atty:
+          yield (cls.TTY_PATH_ENV.format(fd_id), os.ttyname(fd.fileno()) or '')
```

Comment:
Is an empty value useful here? If not, maybe phrase this as:

```
tty_path = os.ttyname(fd.fileno())
if tty_path:
  yield (cls.TTY_PATH_ENV.format(fd_id), tty_path)
```
?

Reply:
I think if the fd is a tty, I would always expect this env var to be populated for clean retrieval on the other side - so yeah.